crates/primitives/src/header.rs

```diff
@@ -186,8 +198,17 @@ impl Encodable for Header {
         self.extra_data.encode(out);
         self.mix_hash.encode(out);
         H64::from_low_u64_be(self.nonce).encode(out);
+
+        // Encode base fee. Put empty string if base fee is missing,
+        // but withdrawals root is present.
         if let Some(ref base_fee) = self.base_fee_per_gas {
             U256::from(*base_fee).encode(out);
+        } else if self.withdrawals_root.is_some() {
+            out.put_u8(EMPTY_STRING_CODE);
+        }
```

Comment:
i thought so as well, but there is a conflict between encodings. `main_codec` requires that the `Bytes` (`extra_data`) type is placed last since it doesn't have known size. but in order to derive rlp en/de, we would need to rearrange the field order and have base fee and withdrawals root last